optuna/integration/cma.py

```diff
@@ -277,24 +269,14 @@ def _initialize_sigma0(search_space: Dict[str, BaseDistribution]) -> float:
 
         sigma0s = []
         for name, distribution in search_space.items():
-            if isinstance(distribution, UniformDistribution):
```

Comment:
I think this change (for `FloatDistribution`) should not be included in this PR. Could you please extract changes for `IntDistribution`?

Reply:
Thank you for pointing out, sorry while resolving conflicts I went ahead for some changes involves FloatDistribution but will revert it. Some parts of change may conflict with works on FloatDistribution side (e.g. below numbered parametrize), but I understood that we can resolve it later. Thank you for the guidance, will work on it by tomorrow.

https://github.com/optuna/optuna/commit/c6663a1f4ddf11039eee8743bac2924598a660a9#diff-8b85ccf8a3534b2aa878bce8ece8ce930af410ab178ebc6d80031badbabb5052L842-R818

Reply:
Thank you for the quick response and your cooperation @nyanhi. ❤️
This refactoring is complicated and I force you to revise for resolving them, I really appreciate your contribution.

The same refactoring on `FloatDistribution` side is going on by @xadrianzetx.
We will merge this PR (#3181) before #3275, so we will ask @xadrianzetx to resolve conflict if it is needed.
In this PR, please focus on cleaning up `IntDistribution` and its subclasses. :pray: